projects/aws/bottlerocket-bootstrap/pkg/kubeadm/utils.go

```diff
@@ -104,6 +105,71 @@ func getBootstrapFromJoinConfig(path string) (string, string, error) {
 	return "https://" + serverEndpoint, token, nil
 }
 
+func getLocalApiServerBindPortFromInitConfig(path string) (int, error) {
+	data, err := ioutil.ReadFile(path)
+	if err != nil {
+		return 0, errors.Wrap(err, "Error reading kubeadm init config file")
+	}
+	return getLocalApiBindPortFromInitConfigYaml(string(data))
+}
+
+func getLocalApiBindPortFromInitConfigYaml(yamlString string) (int, error) {
+	yamlTrees, err := utils.UnmarshalYamlIntoMaps(yamlString)
+	if err != nil {
+		return 0, err
+	}
+
+	var kubeadmInitData map[string]interface{}
+	for _, yamlTree := range yamlTrees {
+		kind, ok := yamlTree["kind"]
+		if ok && kind == "InitConfiguration" {
+			kubeadmInitData = yamlTree
+		}
+	}
+	if kubeadmInitData == nil {
+		return 0, errors.New("Cannot find InitConfiguration")
+	}
+
+	localAPIEndpoint := kubeadmInitData["localAPIEndpoint"].(map[string]interface{})
+	bindPort := int(localAPIEndpoint["bindPort"].(float64))
+
+	return bindPort, nil
+}
+
+func getLocalApiBindPortFromJoinConfig(path string) (int, error) {
+	data, err := ioutil.ReadFile(path)
+	if err != nil {
+		return 0, errors.Wrap(err, "Error reading kubeadm join config file")
```

Comment:
```suggestion
		return 0, errors.Wrap(err, "read kubeadm join config file")
```

Reply:
How about `return 0, errors.Wrap(err, "failed to read kubeadm join config file")`?

Reply:
See https://github.com/aws/eks-anywhere-build-tooling/pull/840#discussion_r880387197.

You decide.